packages/block-editor/src/components/inserter/hooks/use-patterns-paging.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo, useState } from '@wordpress/element';
+
+import { useAsyncList, usePrevious } from '@wordpress/compose';
+
+const PAGE_SIZE = 20;
+const INITIAL_INSERTER_RESULTS = 5;
+
+/**
+ * Supplies values needed to page the patterns list client side.
+ *
+ * @param {Array}  currentCategoryPatterns An array of the current patterns to display.
+ * @param {string} currentCategory         The currently selected category.
+ * @param {string} scrollContainerClass    Class of container to scroll when moving between pages.
+ * @param {string} currentFilter           The currently search filter.
+ *
+ * @return {Object} Returns the relevant paging values. (totalItems, categoryPatternsList, numPages, changePage, currentPage)
+ */
+export default function usePatternsPaging(
+	currentCategoryPatterns,
+	currentCategory,
+	scrollContainerClass,
```

Comment:
Possibly a better API would be accepting a `scrollingContainerRef` to call `scrollToTop` from.

Reply:
This would work in the tab, but in the [patterns explorer ](https://github.com/WordPress/gutenberg/pull/54007/files#diff-6b4f2493a44f524f5f43e18b6a3a655fb59d8306e5d981fafd8363f9b96ded9dR119) the scrollable element is the modal components content area and I couldn't see a way to access the ref from that from within patterns-list component - can you see any way that we could get a ref to that scrollable element?

Reply:
Would `wp.dom.getScrollContainer` work?

Reply:
Nice! that works, so I have [switched to using a ref with that method](https://github.com/WordPress/gutenberg/pull/54007/commits/2f53ea25465d4bb9a1911f0aa678fdfbef408257). Thanks.